Readme.md

```diff
@@ -127,62 +113,18 @@ final class ConfirmOrder
 }
 ```
 
-### CommandHandler
-
-```php
-/**
- * @Flow\Scope("singleton")
- */
-class ButtonCommandHandler
-{
-    /**
-     * @var ButtonRepository
-     * @Flow\Inject
-     */
-    protected $buttonRepository;
-
-    /**
-     * @param CreateButton $command
-     */
-    public function handleCreateButton(CreateButton $command)
-    {
-        $button = new Button($command->getIdentifier(), $command->getPublicIdentifier());
-        $button->changeLabel($command->getLabel());
-
-        $this->buttonRepository->save($button);
-    }
-}
-```
-
-### Monitoring
-
-You can disable (enabled by default) the command handler performance monitoring. The monitoring is implemented with AOP,
-check ```Settings.yaml``` for the configuration.
-
-## Domain
-
-* [x] **AggregateRoot**: implement your own based on ```AggregateRootInterface``` with ```AggregateRootTrait```
-* [x] **Repository**: implement your own based on ```RepositoryInterface```
-
-## Event
-
-### EventBus
-
-* [x] **EventBus**: default implementation, implement your own based on ```EventBusInterface```
-
-### EventListenerLocator
-
-* [x] **EventListenerLocator**: based on convention, implement your own based on ```EventListenerLocatorInterface```
+Notice that the commands do not have any dependency on the framework. They are pure POJOs and
+should optimally be immutable.
 
 ### Event
 
-* [x] **Event**: implement your own based on ```EventInterface```
-
-This interface contains no methods, so you are free to focus on your domain. The interface is used by Flow to provide
-infrastructure helpers (monitoring, debugging, ...).
+Events model a fact that something happened in your domain that is of interest. They should be fully self-contained
+with all information that is necessary to derive a meaningful interpretation of the fact and be immutable.
+Events need to implement the ```EventInterface``` marker Interface, which contains no methods, so you are free to focus on your domain.
+The interface is used by Flow to provide infrastructure helpers (monitoring, debugging, ...).
 
 ```php
-class ProductedOrdered implements EventInterface
+final class ProductedOrdered implements EventInterface
```

Comment:
Lets call this `ProductWasOrdered`. This makes the listener methods much easier to read (`whenProductWasOrdered`)